### PR TITLE
refactor(lb/pool): update lb_pool and lb_member with v2 api

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

update lb_pool and lb_member with v2 api

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes NONE

**Special notes for your reviewer**:

**Release note**:

```release-note
update lb_pool and lb_member with v2 api
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccLBV2Pool_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccLBV2Pool_basic -timeout 360m -parallel 4
=== RUN   TestAccLBV2Pool_basic
=== PAUSE TestAccLBV2Pool_basic
=== CONT  TestAccLBV2Pool_basic
--- PASS: TestAccLBV2Pool_basic (240.50s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       241.687s

make testacc TEST='./huaweicloud' TESTARGS='-run TestAccLBV2Member_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccLBV2Member_basic -timeout 360m -parallel 4
=== RUN   TestAccLBV2Member_basic
=== PAUSE TestAccLBV2Member_basic
=== CONT  TestAccLBV2Member_basic
--- PASS: TestAccLBV2Member_basic (279.88s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       280.115s
```

### CODE REVIEW

File: huaweicloud/lb_v2_shared.go [hunk 1]
```diff
@@ -280,7 +281,7 @@ func waitForLBV2viaPool(networkingClient *golangsdk.ServiceClient, id string, ta
 	if pool.Loadbalancers != nil {
 		// each pool has an LB in Octavia lbaasv2 API
 		lbID := pool.Loadbalancers[0].ID
-		return waitForLBV2LoadBalancer(networkingClient, lbID, target, nil, timeout)
+		return waitForLBV2LoadBalancer_v2(networkingClient, lbID, target, nil, timeout)
```
Comment (chengxiangdong) [inline huaweicloud/lb_v2_shared.go:284]: `waitForLBV2LoadBalancer_v2` is not a good function name, it's up to you to rename it or not

File: huaweicloud/resource_huaweicloud_lb_member.go [hunk 2]
```diff
@@ -190,12 +200,12 @@ func resourceMemberV2Update(d *schema.ResourceData, meta interface{}) error {
...
 	err = waitForLBV2viaPool(lbClient, poolID, "ACTIVE", timeout)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	logp.Printf("[DEBUG] Updating member %s with options: %#v", d.Id(), updateOpts)
 	//lintignore:R006
-	err = resource.Retry(timeout, func() *resource.RetryError {
+	err = resource.RetryContext(ctx, timeout, func() *resource.RetryError {
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_member.go:208]: does it need to retry after API changes?

File: huaweicloud/resource_huaweicloud_lb_member.go [hunk 3]
```diff
@@ -204,35 +214,35 @@ func resourceMemberV2Update(d *schema.ResourceData, meta interface{}) error {
...
 	err = waitForLBV2viaPool(lbClient, poolID, "ACTIVE", timeout)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	logp.Printf("[DEBUG] Attempting to delete member %s", d.Id())
 	//lintignore:R006
-	err = resource.Retry(timeout, func() *resource.RetryError {
+	err = resource.RetryContext(ctx, timeout, func() *resource.RetryError {
```
Comment (chengxiangdong) [inline huaweicloud/resource_huaweicloud_lb_member.go:245]: ditto

File: huaweicloud/resource_huaweicloud_lb_member.go (hunk 2 above)
Comment (ShiChangkuo) [reply huaweicloud/resource_huaweicloud_lb_member.go:208]: @chengxiangdong just keep it as is, will analyze it later.

Comment (chengxiangdong) [reply huaweicloud/resource_huaweicloud_lb_member.go:208]: ok

**Pull request merged**